library/platform_util.c

```diff
@@ -168,13 +168,34 @@ uint32_t mbedtls_platform_random_in_range( size_t num )
 #endif
 }
 
-/* Some compilers (armcc 5 for example) optimize away successive reads from a
- * volatile local variable (which we use as a counter-measure to fault
- * injection attacks), unless there is a call to an external function between
- * them. This functions doesn't need to do anything, it just needs to be
- * in another compilation unit. So here's a function that does nothing. */
+int mbedtls_platform_random_delay( size_t max_rand )
+{
+#if !defined(MBEDTLS_ENTROPY_HARDWARE_ALT)
+    (void) max_rand;
+    return -1;
+#else
+    size_t random_number;
+    volatile size_t i = 0;
+    if( max_rand == 0 || max_rand > INT_MAX )
+    {
+        return( -1 );
+    }
+
+    random_number = mbedtls_platform_random_in_range( max_rand );
+
+    do
+    {
+        i++;
```

Comment:
Major: We should do some random work here, not just increment a counter. In a power trace, it probably looks like a uniform delay, and may easily be removed from the trace by analysis.

We should extract two random `uint32` words, and xor themselves together repeatedly. That will show variation in the power trace on each iteration of the loop.

Reply:
Thanks @sbutcher-arm , this is fixed in the commit "_Add more variation to random delay countermeasure_"